File: 350. Intersection of Two Arrays II/350_test.go
package leetcode

import(
	"testing"
	"reflect"
)

func TestIntersect(t *testing.T){
	type test struct{
		input1 []int
		input2 []int
		want []int
	}
	
	tests := map[string]test{
	"1": test{[]int{1,2,2,1}, []int{2,2}, []int{2,2}},
	"2": test{[]int{4,9,5}, []int{9,4,9,8,4}, []int{9, 4}},
	}
	
	for name, tc := range tests{
		t.Run(name, func(t *testing.T){
			got := intersect(tc.input1, tc.input2)
			if !reflect.DeepEqual(got, tc.want){
				t.Errorf("Test Failed! want: %#v got: %#v", tc.want, got)
			}
		})
	}
}

func BenchmarkIntersect(b *testing.B){
	for i:=0; i<b.N; i++{
		intersect([]int{4,9,5}, []int{9,4,9,8,4})
	}
}

File: 350. Intersection of Two Arrays II/_350_test.py
import unittest
from _350 import *

class test():
    def __init__(self, input1, input2, want):
        self.input1 = input1
        self.input2 = input2
        self.want = want


class myTest(unittest.TestCase):
    def testIntersect(self):
        so = Solution()

        test1 = test([1,2,2,1], [2,2], [2,2])
        test2 = test([4,9,5], [9,4,9,8,4], [9,4])

        got1 = so.intersect(test1.input1, test1.input2)
        got2 = so.intersect(test2.input1, test2.input2)

        self.assertEqual(test1.want, got1, "Test1 Failed!")
        self.assertEqual(test2.want, got2, "Test2 Failed!")

if __name__ == "__main__":
    testSuite = unittest.TestSuite()
    testSuite.addTest(myTest("testIntersect"))
    unittest.TextTestRunner(testSuite)